docs/docs/basics/types.md

```diff
@@ -2,89 +2,224 @@
 
-```haskell
-False :: Bool
-```
+Similarly, these are expressions for the value `False`, which also has the type `Bool`.
+
+=== "In a repl"
+
+    ```haskell
+    False :: Bool
+    False && True :: Bool
+    False || False :: Bool
+    ```
+
+=== "In a file"
+
+    ```haskell
+    example4 :: Bool
+    example4 = False
+
+    example5 :: Bool
+    example5 = False && True
+
+    example6 :: Bool
+    example6 = False || False
+    ```
+
+These expressions makes an explicit statement about the type, and it's written that way above to demonstrate the relationship between values and types, and show the meaning of `::`, which can be read as "has the type". In general, though, you are not required to include the type every time you use a value in Haskell. Because it's optional but can be attached to a value for clarity, `:: Bool` is called a *type* *annotation*.
+
+In the REPL, you can also ask for the type of an expression using `:t`, like this:
+
+=== "In a repl"
+
+    ```haskell
+    > :t True
+    True :: Bool
+    
+    > :t False
+    False :: Bool
+    
+    > :t True && False
+    True && False :: Bool
+    ```
 
 !!! Note
-    In Haskell, everything from simple values like `True` to complex programs have a unique type. 
+    In Haskell, every value, from a simple value like `True` to a complex program, has a type.
 
 !!! Tip
-    Haskell types can be quite complex. To understand a type, always ask: what do the values belonging to this type look like?
+    'Bool' is a simple type, but Haskell types can become quite complex, too. To understand a type, always start by asking: what do the values belonging to this type look like?
 
     For example, the values belonging to `Bool` are `True` and `False`.
 
-## The type of integers
+## Types of integers
 
-`Int` is a type for integers, as in:
+`Int` is one type for integers. An `Int` is a signed integer with a fixed number of bits, but the exact number of bits depends on the Haskell implementation and architecture.
 
-```haskell
-5 :: Int
-```
+=== "In a repl"
+
+    ```haskell
+    5 :: Int
+    ```
+
+=== "In a file"
+
+    ```haskell
+    x :: Int
+    x = 5
+    ```
+
+Sometimes you might want to work with integers whose values can be arbitrarily large.  For this, there's another type: `Integer`. The same expression, `5`, can be used to describe an `Integer`, as well!
+
+=== "In a repl"
+
+    ```haskell
+    5 :: Integer
+    5 * 1000000000000000000 :: Integer
+    ```
+
+=== "In a file"
+
+    ```haskell
+    y1 :: Integer
+    y1 = 5
+
+    y2 :: Integer
+    y2 = 5 * 1000000000000000000
+    ```
+
+An expression like `5` that can more than one type is said to be *overloaded*. It can describe different values depending on the type it's used as.
```

Comment:
Given the subsequent Gotcha section, and the part you added, I think this is redundant.